### ISSUE TITLE
Custom index code not covered by simplecov

### ISSUE THREAD

**Issue by thisismydesign** [activeadmin/activeadmin#5954]
### Expected behavior

When using `simplecov` I expect tested admin code to marked as covered.

### Actual behavior

Anything inside 'Index as a Table' block is not covered even though the code is verifiably hit. Other customizations such as `show` and `form` work well.

### How to reproduce

`app/admin/users.rb`
```
ActiveAdmin.register User do
  index do
    selectable_column

    User.column_names.each do |c|
      pp 'NOTICE ME'
      column c.to_sym unless c.ends_with?('_id') || c.ends_with?('_count')
    end

    User.reflect_on_all_associations(:belongs_to).map(&:name).each do |association|
      column association
    end

    actions
  end
end
```

`spec/controllers/admin/users_controller.rb`
```
require 'rails_helper'

RSpec.describe Admin::UsersController, type: :controller do
  render_views

  before do
    create(:users)
  end

  describe 'GET /' do
    it 'reponds with success' do
      get :index
      expect(response).to be_successful
    end
  end
end
```

- notice prints of `NOTICE ME` in the console
- `open coverage/index.html`
- notice missing coverage for the `index` block

### Notes

Note that `show` and `form` work fine so please don't dismiss this as an issue with `simplecov` without explaining what `index` is doing differently. I think it's fair to expect not too much magic so that least test coverage works, which is why I'm reporting this issue here instead of `simplecov`.

**Comment by deivid-rodriguez** [comment]
Thanks for the report @thisismydesign, I'll try to figure out what's going on here.

**Comment by gonuguntla** [comment]
Please can you share the rails_helper file for active-admin. I am trying write the test cases for active-admin, test cases are not hitting the active-admin code.



  require 'rails_helper'

  RSpec.describe Admin::FaqsController, :type => :controller do
    render_views

    let(:faq) { Capybara::Node::Simple.new(response.body) } 

      let!(:faq) { Faq.create(id: 1) }

    let!(:faq) { Fabricate :faq }

    let(:valid_attributes) do
      Fabricate.attributes_for :faq
    end


    describe "POST create" do
      it 'create an faq' do
        post :create, faq: { id: 1, question: "test@email.com", answer: "ss",  position: 1 }
        expect(Faq.all.count).to eq(1)
      end
    end

      describe "GET show" do
        before do
          get :show, faq: { id: 1, question: "test@email.com", answer: "ss",  position: 1 }
        end
      end
  end



# This file is copied to spec/ when you run 'rails generate rspec:install'
require 'spec_helper'
require 'simplecov'
#if ENV['coverage'] == 'on'
 SimpleCov.start 'rails' do
  add_filter '/bin/'
  add_filter '/db/'
  add_filter '/test/' # for minitest
end
# fail unit tests if total coverage dips below acceptable limit
SimpleCov.minimum_coverage 50
# fail unit tests if any file's individual coverage dips below acceptable limit
SimpleCov.minimum_coverage_by_file 0
#end

ENV['RAILS_ENV'] ||= 'test'
require 'spec_helper'
require File.expand_path('../../config/environment', __FILE__)
require 'rspec/rails'
require 'shoulda/matchers'

# Requires supporting ruby files with custom matchers and macros, etc, in
# spec/support/ and its subdirectories. Files matching `spec/**/*_spec.rb` are
# run as spec files by default. This means that files in spec/support that end
# in _spec.rb will both be required and run as specs, causing the specs to be
# run twice. It is recommended that you do not name files matching this glob to
# end with _spec.rb. You can configure this pattern with with the --pattern
# option on the command line or in ~/.rspec, .rspec or `.rspec-local`.
Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }

# Checks for pending migrations before tests are run.
# If you are not using ActiveRecord, you can remove this line.
ActiveRecord::Migration.maintain_test_schema!

RSpec.configure do |config|
  # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
  config.fixture_path = "#{::Rails.root}/spec/fixtures"

  # If you're not using ActiveRecord, or you'd prefer not to run each of your
  # examples within a transaction, remove the following line or assign false
  # instead of true.
  config.use_transactional_fixtures = true

  # RSpec Rails can automatically mix in different behaviours to your tests
  # based on their file location, for example enabling you to call `get` and
  # `post` in specs under `spec/controllers`.
  #
  # You can disable this behaviour by removing the line below, and instead
  # explicitly tag your specs with their type, e.g.:
  #
  #     RSpec.describe UsersController, :type => :controller do
  #       # ...
  #     end
  #
  # The different available types are documented in the features, such as in
  # https://relishapp.com/rspec/rspec-rails/docs
  config.infer_spec_type_from_file_location!
  config.include Devise::Test::ControllerHelpers, type: :controller

  #config.include FactoryGirl::Syntax::Methods
end

**Comment by dtbaker** [comment]
Has anyone figured this out yet?

simplecov isn't covering this part for me either

![image](https://user-images.githubusercontent.com/543073/220043554-8b6f959c-322b-42e9-ace4-ac4b1f0b7968.png)

**Comment by dtbaker** [comment]
Yes we have test data, seeded with factorybot, as the spec is verifying that the expected data appears in these columns.

**Comment by pc-orion** [comment]
I'm still getting the same issue index part not covered. If someone have any solution please share.
